ocaml/xapi/repository.ml

```diff
@@ -178,14 +180,16 @@ let sync ~__context ~self ~token ~token_id =
           ; "--downloadcomps"
           ; "--download-metadata"
           ; "--delete"
-          ; "--plugins"
           ; "--newest-only"
           ; Printf.sprintf "--repoid=%s" repo_name
           ]
+          @ plugins ()
+          |> add_sub_cmd Reposync
         in
+
         Unixext.mkdir_rec !Xapi_globs.local_pool_repo_dir 0o700 ;
         clean_yum_cache repo_name ;
-        ignore (Helpers.call_script !Xapi_globs.reposync_cmd sync_params)
+        ignore (Helpers.call_script (reposync_cmd ()) sync_params)
```

Comment:
Generally, could all these parameters be wrapped into `Pkg_mgr`? So that the interface would be as simple as e.g. `Pkg_mgr.sync_repo` , etc.

Reply:
The point here is `Pkg_mrg` does NOT want to handle the details output logics, 
What it does is just dispatch the command line and arguments.